fsw/src/generic_css_app.c

```diff
@@ -0,0 +1,614 @@
+                GENERIC_CSS_Disable();
+            }
+            break;
+
+        /*
+        ** TODO: Edit and add more command codes as appropriate for the application
+        ** Set Configuration Command
+        ** Note that this is an example of a command that has additional arguments
+        */
+        /*
+        case GENERIC_CSS_CONFIG_CC:
+            if (GENERIC_CSS_VerifyCmdLength(GENERIC_CSS_AppData.MsgPtr, sizeof(GENERIC_CSS_Config_cmd_t)) == OS_SUCCESS)
+            {
+                CFE_EVS_SendEvent(GENERIC_CSS_CMD_CONFIG_INF_EID, CFE_EVS_INFORMATION, "GENERIC_CSS: Configuration command received");
+                // Command device to send HK
+                status = GENERIC_CSS_CommandDevice(GENERIC_CSS_AppData.Generic_cssUart.handle, GENERIC_CSS_DEVICE_CFG_CMD, ((GENERIC_CSS_Config_cmd_t*) GENERIC_CSS_AppData.MsgPtr)->DeviceCfg);
+                if (status == OS_SUCCESS)
+                {
+                    GENERIC_CSS_AppData.HkTelemetryPkt.DeviceCount++;
+                }
+                else
+                {
+                    GENERIC_CSS_AppData.HkTelemetryPkt.DeviceErrorCount++;
+                }
+            }
+            break;
+        */
+        /*
+        ** Invalid Command Codes
+        */
+        default:
+            /* Increment the error counter upon receipt of an invalid command */
+            GENERIC_CSS_AppData.HkTelemetryPkt.CommandErrorCount++;
+            CFE_EVS_SendEvent(GENERIC_CSS_CMD_ERR_EID, CFE_EVS_ERROR, 
+                "GENERIC_CSS: Invalid command code for packet, MID = 0x%x, cmdCode = 0x%x", MsgId, CommandCode);
+            break;
+    }
+    return;
+} 
+
+
+/*
+** Process Telemetry Request - Triggered in response to a telemetery request
+** TODO: Add additional telemetry required by the specific component
+*/
+void GENERIC_CSS_ProcessTelemetryRequest(void)
+{
+    int32 status = OS_SUCCESS;
+
+    /* MsgId is only needed if the command code is not recognized. See default case */
+    CFE_SB_MsgId_t MsgId = CFE_SB_GetMsgId(GENERIC_CSS_AppData.MsgPtr);   
+
+    /* Pull this command code from the message and then process */
+    uint16 CommandCode = CFE_SB_GetCmdCode(GENERIC_CSS_AppData.MsgPtr);
+    switch (CommandCode)
+    {
+        case GENERIC_CSS_REQ_HK_TLM:
+            //GENERIC_CSS_ReportHousekeeping();
+            //break;
+
+        case GENERIC_CSS_REQ_DATA_TLM:
+            GENERIC_CSS_ReportDeviceTelemetry();
+            break;
+
+        /*
+        ** Invalid Command Codes
+        */
+        default:
+            /* Increment the error counter upon receipt of an invalid command */
+            GENERIC_CSS_AppData.HkTelemetryPkt.CommandErrorCount++;
+            CFE_EVS_SendEvent(GENERIC_CSS_DEVICE_TLM_ERR_EID, CFE_EVS_ERROR, 
+                "GENERIC_CSS: Invalid command code for packet, MID = 0x%x, cmdCode = 0x%x", MsgId, CommandCode);
+            break;
+    }
+    return;
+}
+
+
+/* 
+** Report Application Housekeeping
```

Comment:
Application HK should always be reported no matter the state (enabled / disabled) so we can just remove the if check for it.